bundles/binding/org.openhab.binding.sonance/src/main/java/org/openhab/binding/sonance/internal/SonanceBinding.java

```diff
@@ -0,0 +1,447 @@
+	 * through the ConfigAdmin service.
+	 * 
+	 * @param configuration
+	 *            Updated configuration properties
+	 */
+	public void modified(final Map<String, Object> configuration) {
+		String refreshIntervalString = (String) configuration.get("refresh");
+		if (StringUtils.isNotBlank(refreshIntervalString))
+			refreshInterval = Long.parseLong(refreshIntervalString);
+	}
+
+	/**
+	 * Called by the SCR to deactivate the component when either the
+	 * configuration is removed or mandatory references are no longer satisfied
+	 * or the component has simply been stopped.
+	 * 
+	 * @param reason
+	 *            Reason code for the deactivation:<br>
+	 *            <ul>
+	 *            <li>0 – Unspecified
+	 *            <li>1 – The component was disabled
+	 *            <li>2 – A reference became unsatisfied
+	 *            <li>3 – A configuration was changed
+	 *            <li>4 – A configuration was deleted
+	 *            <li>5 – The component was disposed
+	 *            <li>6 – The bundle was stopped
+	 *            </ul>
+	 */
+	public void deactivate(final int reason) {
+		// deallocate resources here that are no longer needed and
+		// should be reset when activating this binding again
+		socketCache.clear();
+		outputStreamCache.clear();
+		bufferedReaderCache.clear();
+	}
+
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	@Override
+	protected String getName() {
+		return "Sonance Refresh Service";
+	}
+
+	@Override
+	protected void execute() {
+		if (!bindingsExist()) {
+			logger.debug("There is no existing Sonance binding configuration => refresh cycle aborted!");
+			return;
+		}
+
+		logger.info("Refreshing all items");
+
+		List<String> offlineEndPoints = new ArrayList<String>();
+
+		for (SonanceBindingProvider provider : providers) {
+			for (String itemName : provider.getItemNames()) {
+				String group = provider.getGroup(itemName);
+				String ip = provider.getIP(itemName);
+				int port = provider.getPort(itemName);
+
+				String key = ip + ":" + port;
+				if (!offlineEndPoints.contains(key)) {
+					try {
+						if (!socketCache.containsKey(key)) {
+							socketCache.put(key, new Socket(ip, port));
+							outputStreamCache.put(key, new DataOutputStream(
+									socketCache.get(key).getOutputStream()));
+							bufferedReaderCache.put(key, new BufferedReader(
+									new InputStreamReader(socketCache.get(key)
+											.getInputStream())));
+							logger.debug("New socket created");
+						}
+
+						if (provider.isMute(itemName))
+							sendMuteCommand(itemName, "FF550212" + group,
+									outputStreamCache.get(key),
+									bufferedReaderCache.get(key));
+						else if (provider.isVolume(itemName))
+							sendVolumeCommand(itemName, "FF550210" + group,
+									outputStreamCache.get(key),
+									bufferedReaderCache.get(key));
+					} catch (UnknownHostException e) {
+						logger.error("UnknownHostException occured.");
```

Comment:
which host is unknown? Please insert meaningful logging … an error has occurred is not helpful when debugging this binding.